zerver/lib/message.py

```diff
@@ -196,6 +197,23 @@ def normalize_body(body: str) -> str:
     return truncate_content(body, settings.MAX_MESSAGE_LENGTH, "\n[message truncated]")
 
 
+def normalize_body_for_import(messages: list[Record]) -> list[Record]:
+    for message in messages:
+        content = message["content"]
+        rendered_content = message["rendered_content"]
+        if "\x00" in content:
+            content = re.sub(r"\x00", "", content)
+        if "\x00" in rendered_content:
+            rendered_content = re.sub(r"\x00", "", rendered_content)
+        content = truncate_content(content, settings.MAX_MESSAGE_LENGTH, "\n[message truncated]")
+        rendered_content = truncate_content(
+            rendered_content, settings.MAX_MESSAGE_LENGTH, "\n[message truncated]"
+        )
+        message["content"] = content
+        message["rendered_content"] = rendered_content
```

Comment:
Why does this mutate `rendered_content` too? I would expect us to be calling this in the export tools, not the import tools, before `rendered_content` is computed from `content`.

Reply:
Yes, at this stage we haven't rendered the third-party exported message yet. in fact, we explicitly set it to `None` in `channel_message_to_zerver_message`. For third-party exports, they'll get rendered during import. So I don't think we should do that here too

Reply:
moved it to `import_util.py`, ensuring null bytes are removed before `rendered_content` is computed. please check